include/crud-terms.php

```diff
@@ -307,25 +276,52 @@ private function get_slug_separator() {
 	 * @return string Slug with a language suffix if found.
 	 */
 	public function set_pre_term_slug( $slug, $taxonomy ) {
-		if ( ! empty( $slug ) ) {
-			return $slug;
+		if ( ! $slug ) {
+			$slug = sanitize_title( $this->pre_term_name );
 		}
 
-		$name = sanitize_title( $this->pre_term_name );
+		if ( ! $this->model->is_translated_taxonomy( $taxonomy ) || ! term_exists( $slug, $taxonomy ) ) {
+			return $slug;
+		}
 
 		/**
 		 * Filters the subsequently inserted term language.
 		 *
 		 * @since 3.3
 		 *
 		 * @param PLL_Language|null $lang     Found language object, null otherwise.
+		 * @param string            $taxonomy Term taonomy.
 		 * @param string            $slug     Term slug
-		 * @param string            $taxonomy Term taxonomy.
 		 */
-		$lang = apply_filters( 'pll_inserted_term_language', null, $name, $taxonomy );
+		$lang = apply_filters( 'pll_inserted_term_language', null, $taxonomy, $slug );
 
-		if ( $lang instanceof PLL_Language && ! $this->model->term_exists_by_slug( $name, $lang, $taxonomy ) ) {
-			$slug = $name . $this->get_slug_separator() . $lang->slug;
+		if ( $lang instanceof PLL_Language ) {
+			/**
+			 * Filters the subsequently inserted term parent.
+			 *
+			 * @since 3.3
+			 *
+			 * @param int          $parent   Parent term ID, 0 if none.
+			 * @param string       $taxonomy Term taxonomy.
+			 * @param string       $slug     Term slug
+			 */
+			$parent = apply_filters( 'pll_inserted_term_parent', 0, $taxonomy, $slug );
+
+			/**
+			 * Special case if the parent has the same slug.
+			 * WordPress prepends the slug to itself, so let's do it.
+			 */
+			$parent_obj = get_term( $parent, $taxonomy );
+			if ( $parent_obj instanceof WP_Term && $parent_obj->slug === $slug ) {
+				return $slug .= '-' . $slug;
+			}
```

Comment:
Same comment I made [here](https://github.com/polylang/polylang-pro/pull/1490/files#r993631909).

Reply:
Good idea! Done in 6728fe0